src/dataflow/deadlines.rs

```diff
@@ -13,105 +25,78 @@ use tokio::time::Duration;
 pub trait CondFn: Fn(&ConditionContext, &Timestamp) -> bool + Send + Sync {}
 impl<F: Fn(&ConditionContext, &Timestamp) -> bool + Send + Sync> CondFn for F {}
 
-/// A `DeadlineEvent` structure defines a deadline that is generated upon the fulfillment of a
-/// start condition on a given stream and a given timestamp (we assume a single deadline for each
-/// timestamp). Upon expiration of the deadline (defined as `duration`), the `end_condition`
-/// function is checked, and the `handler` is invoked if it was not satisfied.
-pub struct DeadlineEvent {
-    pub stream_id: StreamId,
-    pub timestamp: Timestamp,
-    pub duration: Duration,
-    pub handler: Arc<Mutex<dyn HandlerContextT>>,
-    pub end_condition: Arc<dyn CondFn>,
-}
+/*************************************************************************************************
+ * DeadlineFn: A DeadlineFn defines the duration between the start and the end condition that an *
+ * operator should be restricted to.                                                             *
+ ************************************************************************************************/
 
-impl DeadlineEvent {
-    pub fn new(
-        stream_id: StreamId,
-        timestamp: Timestamp,
-        duration: Duration,
-        handler: Arc<Mutex<dyn HandlerContextT>>,
-        end_condition: Arc<dyn CondFn>,
-    ) -> Self {
-        Self {
-            stream_id,
-            timestamp,
-            duration,
-            handler,
-            end_condition,
-        }
-    }
-}
+/// A trait that defines the deadline function.
```

Comment:
Could you describe the role of the deadline function? E.g. maps operator state and timestamp to a time point in the future?